src/oneiro/pipelines/civitai_checkpoint.py

```diff
@@ -777,15 +783,20 @@ def generate(
 
         # Build generation kwargs
         gen_kwargs: dict[str, Any] = {
-            "prompt": prompt,
             "num_inference_steps": steps,
             "guidance_scale": guidance_scale,
             "generator": generator,
         }
 
-        # Add negative prompt only for pipelines that support it
-        if self._pipeline_config.supports_negative_prompt and negative_prompt:
-            gen_kwargs["negative_prompt"] = negative_prompt
+        # Use embedding-based prompt handling for pipelines that support it
+        # (SD 1.x, SD 2.x, SDXL, Flux, SD3) - enables weight syntax like (word:1.5)
+        if self._supports_prompt_embeddings():
+            self._encode_prompts_to_embeddings(gen_kwargs, prompt, negative_prompt)
```

Comment:
@dobbyphus i think this should be something like:

```
gen_kwargs.update(self._encode_prompts_to_embeddings(prompt, negative_prompt))
```

then `_encode_prompts_to_embeddings` just returns a `dict` of kwargs itself, instead of mutating something it doesn't "own"